src/java.base/share/classes/sun/security/ssl/SSLEngineInputRecord.java

```diff
@@ -263,6 +263,10 @@ private Plaintext[] decodeInputRecord(ByteBuffer packet)
         // parse handshake messages
         //
         if (contentType == ContentType.HANDSHAKE.id) {
+            if (!fragment.hasRemaining()) {
```

Comment:
It may be easier to read if using "contentLen != 0", instead of "fragment.hasRemaining()".  I may add a comment like what is says in RFC 8446, "Implementations MUST NOT send zero-length fragments of Handshake types, even if those fragments contain padding."

Reply:
I updated the if-logic and added the comment